@jules 0s
0s
Run cargo fmt --check
Diff in /home/runner/work/do-movie-radio-play/do-movie-radio-play/src/verification/analysis.rs:299:
         let (entropy, _, _, _, _) = compute_spectral_features(&samples).unwrap();
         // For 512 bins (next_power_of_2 of 1024 is 1024, but realfft gives 513 bins),
         // max entropy is log2(513) approx 9.0.
-        assert!(entropy > 7.0, "White noise should have high entropy, got {}", entropy);
+        assert!(
+            entropy > 7.0,
+            "White noise should have high entropy, got {}",
+            entropy
+        );
     }
 
     #[test]
Diff in /home/runner/work/do-movie-radio-play/do-movie-radio-play/src/verification/analysis.rs:320:
         let (_, flatness_sine, _, _, _) = compute_spectral_features(&sine).unwrap();
         let (_, flatness_noise, _, _, _) = compute_spectral_features(&noise).unwrap();
 
-        assert!(flatness_noise > flatness_sine, "Noise flatness {} should be > sine flatness {}", flatness_noise, flatness_sine);
+        assert!(
+            flatness_noise > flatness_sine,
+            "Noise flatness {} should be > sine flatness {}",
+            flatness_noise,
+            flatness_sine
+        );
         // Pure sine has 1 peak, white noise is flat. Flatness of noise should be close to 1.
         // Flatness of a pure sine should be low.
-        assert!(flatness_sine < 0.35, "Sine flatness too high: {}", flatness_sine);
-        assert!(flatness_noise > 0.45, "Noise flatness too low: {}", flatness_noise);
+        assert!(
+            flatness_sine < 0.35,
+            "Sine flatness too high: {}",
+            flatness_sine
+        );
+        assert!(
+            flatness_noise > 0.45,
+            "Noise flatness too low: {}",
+            flatness_noise
+        );
     }
 
     #[test]
+        );
     }
 
     #[test]
Diff in /home/runner/work/do-movie-radio-play/do-movie-radio-play/src/verification/analysis.rs:342:
             samples2[i] = 0.5; // Change second half
         }
         let flux = compute_spectral_flux(&samples2);
-        assert!(flux > 0.0, "Flux should be non-zero for 2 different windows, got {}", flux);
+        assert!(
+            flux > 0.0,
+            "Flux should be non-zero for 2 different windows, got {}",
+            flux
+        );
     }
 }
 
Error: Process completed with exit code 1.

--- a/src/verification/analysis.rs
+++ b/src/verification/analysis.rs
@@ -299,7 +299,11 @@ mod tests {
         let (entropy, _, _, _, _) = compute_spectral_features(&samples).unwrap();
         // For 512 bins (next_power_of_2 of 1024 is 1024, but realfft gives 513 bins),
         // max entropy is log2(513) approx 9.0.
-        assert!(entropy > 7.0, "White noise should have high entropy, got {}", entropy);
+        assert!(
+            entropy > 7.0,
+            "White noise should have high entropy, got {}",
+            entropy
+        );
     }
 
     #[test]
@@ -320,11 +324,24 @@ mod tests {
         let (_, flatness_sine, _, _, _) = compute_spectral_features(&sine).unwrap();
         let (_, flatness_noise, _, _, _) = compute_spectral_features(&noise).unwrap();
 
-        assert!(flatness_noise > flatness_sine, "Noise flatness {} should be > sine flatness {}", flatness_noise, flatness_sine);
+        assert!(
+            flatness_noise > flatness_sine,
+            "Noise flatness {} should be > sine flatness {}",
+            flatness_noise,
+            flatness_sine
+        );
         // Pure sine has 1 peak, white noise is flat. Flatness of noise should be close to 1.
         // Flatness of a pure sine should be low.
-        assert!(flatness_sine < 0.35, "Sine flatness too high: {}", flatness_sine);
-        assert!(flatness_noise > 0.45, "Noise flatness too low: {}", flatness_noise);
+        assert!(
+            flatness_sine < 0.35,
+            "Sine flatness too high: {}",
+            flatness_sine
+        );
+        assert!(
+            flatness_noise > 0.45,
+            "Noise flatness too low: {}",
+            flatness_noise
+        );
     }
 
     #[test]
@@ -342,6 +359,10 @@ mod tests {
             samples2[i] = 0.5; // Change second half
         }
         let flux = compute_spectral_flux(&samples2);
-        assert!(flux > 0.0, "Flux should be non-zero for 2 different windows, got {}", flux);
+        assert!(
+            flux > 0.0,
+            "Flux should be non-zero for 2 different windows, got {}",
+            flux
+        );
     }
 }

--- a/src/verification/analysis.rs
+++ b/src/verification/analysis.rs
@@ -309,8 +309,8 @@ mod tests {
     #[test]
     fn test_spectral_flatness_sine_vs_noise() {
         let mut sine = vec![0.0f32; 1024];
-        for i in 0..1024 {
-            sine[i] = (i as f32 * 0.1).sin();
+        for (i, s) in sine.iter_mut().enumerate() {
+            *s = (i as f32 * 0.1).sin();
         }
 
         let mut noise = vec![0.0f32; 1024];
@@ -355,8 +355,8 @@ mod tests {
         // If we use 0..512 (exclusive), it only sees 1 window.
         // If we use 0..=256, it sees 2 windows.
         let mut samples2 = vec![0.1f32; 768];
-        for i in 512..768 {
-            samples2[i] = 0.5; // Change second half
+        for s in samples2.iter_mut().skip(512) {
+            *s = 0.5; // Change second half
         }
         let flux = compute_spectral_flux(&samples2);
         assert!(

